README.md

```diff
@@ -1,2 +1,2 @@
-# https-checker
-A typechecker for enforcing that URLs start with "https" in Java, implemented as a typesystem for which Strings start with which regexes.
+# StartsWith-checker
+A typechecker for enforcing that URLs start with the given strings in Java, implemented as a typesystem for which Strings start with which regexes.
```

Comment:
This is more general than just for URLs - it could be used for any API that requires Strings to start with certain properties. You should say that here, and then give URLs as one example of such an API.

There's also no support for regexes in this type system - the strings need to be constants.

This README is also just really brief - it would be much improved with a short description of the type system and the subtyping rules.